.github/workflows/rit.yml

```diff
@@ -17,35 +15,134 @@ on:
         required: false
         default: 'master'
 
+env:
+  REGEX_PARSE_BRANCH: '([a-zA-Z0-9/_-]+)'
+  VALID_BRANCH_REGEX: '^[a-zA-Z0-9._/-]+$'
+
 jobs:
   rootstock-integration-tests:
     name: Rootstock Integration Tests
     runs-on: ubuntu-latest
     timeout-minutes: 60
     steps:
+      - name: Checkout Repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
```

Comment:
Is this checkout needed? I don't see anything referenciating the contents.

Reply:
Unfortunately yes, otherwise the next step won't have the github cli available. By using the checkout action, it has everything related with git. At least, I needed when I tried to use the CLI without it.

Reply:
oh right, forgot about the usage of `gh`.

Reply:
Can we add a comment about that here? It isn't obvious.

Reply:
Done